shared_model/interfaces/iroha_internal/proposal.hpp

```diff
@@ -15,15 +15,55 @@
  * limitations under the License.
  */
 
-#ifndef IROHA_PROPOSAL_HPP
-#define IROHA_PROPOSAL_HPP
+#include "interfaces/hashable.hpp"
+#include "interfaces/transaction.hpp"
+#include "model/proposal.hpp"
+
+#ifndef IROHA_SHARED_MODEL_PROPOSAL_HPP
+#define IROHA_SHARED_MODEL_PROPOSAL_HPP
 namespace shared_model {
   namespace interface {
 
-    class Proposal {
-      // TODO implement
+    class Proposal : public Hashable<Proposal, iroha::model::Proposal> {
+      /// Type of proposal transactions' collection
+      using ProposalTransactionsType = std::vector<Transaction>;
+
+      /**
+       * @return collection of proposal's transactions
+       */
+      virtual ProposalTransactionsType &transactions() const = 0;
+
+      /// Type of proposal height
+      using ProposalHeight = uint64_t;
```

Comment:
Does the `ProposalHeight` may be differ e.g from `BlockHeightType`? This alias looks strange to me
ping @muratovv